<launch>
    <!-- Map name is the filename wihtouth the .bt extensionof the 3D Map -->
    <arg name="map_name"            default="mbzirc_challenge3"/>
    <!-- <arg name="map_name"            default="wall"/> -->
    <!-- <arg name="map_name"            default="laberynth"/> -->
    <arg name="map"                 default="$(find heuristic_planners)/resources/3dmaps/$(arg map_name).pcd"/>

    <arg name="algorithm_name"      default="thetastar"/>

    <arg name="world_size_x"        default="60"/>
    <arg name="world_size_y"        default="60"/>
    <arg name="world_size_z"        default="20"/>
    <arg name="resolution"          default="0.1"/>

    <arg name="inflate_map"         default="true"/>
    <arg name="inflation_size"      default="$(arg resolution)"/>
    <!-- Possibles values are: euclidean, euclidean_optimized, manhattan, octogonal and dijkstra -->
    <arg name="heuristic"         default="euclidean"/>

    <arg name="use_marsim"          default="false"/>
    <!-- This should be a folder -->

    <arg name="overlay_markers"     default="false"/>
    
    <arg name="cost_weight" default="4.0"/>
    <arg name="max_line_of_sight_distance" default="2.0"/>

    <!-- GRIDM Generators Parameters -->
    <arg name="cost_scaling_factor" default="2.0"/>  
    <arg name="robot_radius"        default="0.4"/> 
    
    <arg name="output" default="screen"/>
    <node pkg="heuristic_planners" type="planner_ros_node" name="planner_ros_node" output="$(arg output)" required="true" >
        <remap from="points" to="/grid3d/map_point_cloud"/>
        <!-- <remap from="points" to="/map_generator/global_cloud"/> -->
        <!-- <remap from="global_cloud" to="/map_generator/global_cloud"/> -->
        <param name="map_path"              value="$(arg map)"/>
        <param name="world_size_x"          value="$(arg world_size_x)"/>
        <param name="world_size_y"          value="$(arg world_size_y)"/>
        <param name="world_size_z"          value="$(arg world_size_z)"/>
        <param name="resolution"            value="$(arg resolution)"/>
        <param name="inflate_map"           value="$(arg inflate_map)"/>
        <param name="inflation_size"        value="$(arg inflation_size)"/>
        
        <param name="overlay_markers"       value="$(arg overlay_markers)"/>
        <param name="algorithm"             value="$(arg algorithm_name)"/>
        <param name="cost_weight"           value="$(arg cost_weight)"/>
        <param name="max_line_of_sight_distance"   value="$(arg max_line_of_sight_distance)"/>
        <param name="heuristic"             value="$(arg heuristic)"/>

        <param name="cost_scaling_factor"   value="$(arg cost_scaling_factor)"/>
        <param name="robot_radius"          value="$(arg robot_radius)"/>
        <param name="use_marsim"            value="$(arg use_marsim)"/>

        <param name="ground_height_threshold" value="9.0" />
        <param name="ground_to_air_transition_cost" value="0" />
        <param name="air_movement_factor" value="1.0" />
        <param name="flying_cost" value="2.0" />
        <!-- 9.0 / 100 / 1.0 / 1.0 High transition cost -->
        <!-- 9.0 / 0 / 2.0 No Air -->
        <!-- 9.0 / 0 / 1.0 / 2.0 Ideal -->
    </node>

    <node pkg="rviz" name="rviz" type="rviz" args="-d $(find heuristic_planners)/rviz/planners.rviz"/>

    <!-- <node pkg="rqt_service_caller" type="rqt_service_caller" name="rqt_service_caller"/> -->


</launch>
